terraform/modules/statsd/appmesh.tf

```diff
@@ -0,0 +1,49 @@
+resource "aws_appmesh_virtual_service" "statsd" {
```

Comment:
What was the thing that prevented using the app module for this boilerplate stuff? (sorry for being lazy - just easier to ask rather than play spot-the-difference!)

Reply:
The app module sadly uses a bootstrap image, and relies on the app deployments to update the service. Statsd doesn't need an app deployment, so I put it in here for now. I agree the duplication isn't nice - perhaps a mini-module for cloudmap/app mesh might be useful.

Reply:
Oh of course, thanks. I wonder if a small generalisation to the app module might be the way forward - i.e. allow passing in task_definition but have it default to module.bootstrap_task_definition.arn.

Reply:
I considered that, though another problem is that while this service is unlikely to require complex deployment steps or be continuously deployed - so doesn't warrant a deployment - we have [a lifecycle rule](https://github.com/alphagov/govuk-infrastructure/blob/main/terraform/modules/app/main.tf#L55) on the task definition which would require a deployment module. Though perhaps there is a terraform way to wrap this in a conditional?

Reply:
Hm yeah that's a good point, but as you've suggested I think it'd be solved neatly with a straightforward conditional when setting the lifecycle rule.

Reply:
Sadly I have found a conditional isn't possible in a lifecycle rule (https://github.com/hashicorp/terraform/issues/3116#issuecomment-147382638). I'll see if I can find a neat way around this.

Reply:
Oh no 🙈 sorry, didn't mean to send you down a rabbithole there.

Reply:
No it's OK, we were going to hit this problem eventually. A couple of options I'm exploring: a) treat Statsd an app, with an unnecessary deployment module, b) treat Statsd as something different and try to share config with the app module. I've pushed up a change for the second option.